src/test/groovy/transform/stc/GenericsSTCTest.groovy

```diff
@@ -1800,6 +1800,19 @@ assert result == 'ok'
 '''
     }
 
+    // GROOVY-7742
+    void testMethodParamWithDifferentPlaceholderForTypeArgument() {
+        // We use <T> here while List uses <E>
```

Comment:
I don't think using a different placeholder than the generic type is the issue here e.g. if 'T' is replaced with 'U' in this test, it passes without your change.